File: easy/3442_maximum_difference_between_even_and_odd_frequency_i/solution.cpp
#include <string>
#include <vector>

class Solution {
public:
  int maxDifference(std::string s) {
    std::vector<int> frequencies(26, 0);

    for (const char &c : s)
      ++frequencies[c - 'a'];

    int maxOddFrequency = 0;
    int minEvenFrequency = 101;

    for (const int &frequency : frequencies) {
      if (frequency % 2 == 0 && frequency != 0)
        minEvenFrequency = std::min(frequency, minEvenFrequency);
      else
        maxOddFrequency = std::max(frequency, maxOddFrequency);
    }

    return maxOddFrequency - minEvenFrequency;
  }
};

File: easy/3442_maximum_difference_between_even_and_odd_frequency_i/solution.py
from collections import Counter


class Solution:
    def maxDifference(self, s: str) -> int:
        s_counter = Counter(s)

        max_odd_frequency = -1
        min_even_frequency = 101

        for frequency in s_counter.values():
            if frequency % 2 == 1 and frequency > max_odd_frequency:
                max_odd_frequency = frequency
            elif frequency % 2 == 0 and frequency < min_even_frequency:
                min_even_frequency = frequency

        return max_odd_frequency - min_even_frequency
